distribution/tools/plugin-cli/src/test/java/org/elasticsearch/plugins/cli/InstallPluginActionTests.java

```diff
@@ -48,13 +48,17 @@
 import org.elasticsearch.core.Tuple;
 import org.elasticsearch.env.Environment;
 import org.elasticsearch.env.TestEnvironment;
+import org.elasticsearch.plugin.scanner.ClassReadersProvider;
+import org.elasticsearch.plugin.scanner.NamedComponentScanner;
 import org.elasticsearch.plugins.Platforms;
 import org.elasticsearch.plugins.PluginDescriptor;
 import org.elasticsearch.plugins.PluginTestUtil;
 import org.elasticsearch.test.ESTestCase;
 import org.elasticsearch.test.PosixPermissionsResetter;
 import org.junit.After;
 import org.junit.Before;
+import org.mockito.Mockito;
```

Comment:
I don't think we should use mocks here. The rest of the tests here use fakes (we call them mocks, but they are custom mocks/filesystem impls). Ultimately what we are trying to test is just a couple cases, only one or two that need an actual components file, to test the integration.

Reply:
the only way to avoid the use of mockito is to use a "fake jar with no content" so that it relies on nothing being scanned, and nothing being written. (updated the PR)
The classes are being used and file is being created. And this is what that test is mainly trying to show, so perhaps it is ok.